src/static/js/event_animals.js

```diff
@@ -0,0 +1,320 @@
+                        }
+                    },
+                    { field: "IMAGE", display: "", 
+                        formatter: function(row) {
+                            return html.animal_link_thumb_bare(row);
+                        }
+                    },
+                    { field: "ANIMAL", display: _("Animal"), 
+                        formatter: function(row) {
+                            return html.animal_link(row, { noemblems: controller.name == "event_animals" });
+                        },
+                        hideif: function(row) {
+                            // Don't show for animal records
+                            if (controller.animal) { return true; }
+                        }
+                    },
+                    { field: "SPECIESNAME", display: _("Species")
+                    },
+                    { field: "BASECOLOURNAME", display: _("Color")
+                    },
+
+                    { field: "ACCEPTANCENUMBER", display: _("Litter"),
+                        hideif: function(row) {
+                            if (controller.animal) { return true; }
+                            return config.bool("DontShowLitterID");
+                        }
+                    },
+                    { field: "COMMENTS", display: _("Comments")
+                    },
+                    { field: "LASTFOSTERER", display: _("Last Fosterer"),
+                        formatter: function(row) {
+                            return '<span ' +
+                                    (row.LASTFOSTERERRETURNDATE ? 'class="asm-completerow"' : '') + ">" + 
+                                    html.person_link(row.LASTFOSTERERID, row.LASTFOSTERERNAME) + 
+                                    '<br/>' + common.nulltostr(row.LASTFOSTERERMOBILETELEPHONE) + '<br/>' + common.nulltostr(row.LASTFOSTERERHOMETELEPHONE) + '<br/>' + common.nulltostr(row.LASTFOSTERERWORKTELEPHONE) + '</span>';
+                        }
+                    },
+                    { field: "ADOPTED", display: _("Adopted"),
+                        formatter: function(row) {
+                            return row.ADOPTED == 1 ? "&#9989;" : "&nbsp;";
+                        }
+                    }
+                ]
+            };
+
+            const buttons = [
+                    { id: "addanimal", text: _("Add Animal"), icon: "animal-add", tooltip: _("Add animal to this event"), enabled: "always", perm: "cea",
+                        click: async function() { 
+                            $("#addanimal").animalchooser("clear");
+                            await tableform.show_okcancel_dialog("#dialog-addanimal", _("Add"), { notzero: [ "addanimal" ] });
+                            let a = $("#addanimal").animalchooser("get_selected");
+                            await common.ajax_post("event_animals", "mode=create&eventid=" + controller.event.ID + "&animalid=" + a.ID);
+                            //TODO: refresh list
```

Comment:
This is where you should do route_reload() if you want to get the data from the backend again and reload the screen

Reply:
silly me, I found route_reload() but since it didn't work with the previous issue, I didn't try it here.

Reply:
this did the trick!